sips/committee_consensus.md

```diff
@@ -0,0 +1,480 @@
+
+	for i, beaconDuty := range duty.BeaconDuties {
+		validatorPK := types.ValidatorPK(beaconDuty.PubKey)
+		slot, exists := slotMap[validatorPK]
+		if exists {
+			if slot < beaconDuty.Slot {
+				validatorsToStop[beaconDuty.Slot] = validatorPK
+				slot = beaconDuty.Slot
+			} else { // else don't run duty with old slot
+				duty.BeaconDuties[i] = nil
+			}
+		}
+	}
+	return duty, validatorsToStop, slotMap
+}
+```
+2. `Committee` receives consensus messages and hands them over `CommitteeRunner` that hands them over to the `QBFTController` that has unchanged logic. The only difference is `BeaconVote` is used as the ConsensusData object.
+3.  Once `ProcessConsensus` decides, the `Committee` will create a post consensus of PartialSignatureMessage that aggregates the beacon partial signature for all relevant validators.
+4. `Committee` will process post-consensus partial signature messages and submit a beacon message for each validator.
+
+
+### Stopping Runs
+
+Previously we have letted new validator duties stop the run for the previous duty. Now we don't stop the runner but just mark certain validators as stopped and omit the partial sig emission for them.
+
+### Omitting Partial Signatures
+
+If in post-consensus stage for attestation duty, the duty's slot is lower than a validator's `highestAttestingSlot`, the `CommitteeRunner` will omit the post-consesnus message for this validator. 
+
+### Cutoff Round
+We will also create a better Cutoff Round.
+
+#### Sync Committee
+`CutOffRound = 4  \\ one slot`
+
+#### Attestation
+`CutOffRound = 12 \\ one epoch`
+
+### CommitteeID
+
+An identifier for committee must be added to `MessageID`.
+
+```go
+type CommitteeID [32]byte
+
+// Return a 32 bytes ID for the committee of operators
+func getCommitteeID(committee []OperatorID) CommitteeID {
+	// sort
+	sort.Slice(committee, func(i, j int) bool {
+		return committee[i] < committee[j]
+	})
+	// Convert to bytes
+	bytes := make([]byte, len(committee)*4)
+	for i, v := range committee {
+		binary.LittleEndian.PutUint32(bytes[i*4:], uint32(v))
+	}
+	// Hash
+	return CommitteeID(sha256.Sum256(bytes))
+}
+```
+
+In order to route consensus messages to the correct consensus runner, the `CommitteeID` field will be included in the `MessageID` replacing `ValidatorPublicKey`.
+
+### Role
+
+This is used to route the message to the correct runner. Since `CommitteeRunner` has several beacone roles we will create a new `RunnerRole` in the MessageID.
+
+```go
+type RunnerRole int32
+
+const (
+	RoleCommittee RunnerRole = iota
+	RoleAggregator
+	RoleProposer
+	RoleSyncCommitteeContribution
+
+	RoleValidatorRegistration
+	RoleVoluntaryExit
+
+	RoleUnknown = -1
+)
+```
+
+### MessageID
+
+`ValidatorPublicKey` and `CommitteeID` will be used interchangeably in `MessageID`. `Role` will change location because it is used to determine ID type.  `CommitteeID` is 32 bytes long so it will be encoded with a `0x00` 16 bytes prefix to make it the same length as `ValidatorPublicKey`.
```

Comment:
ValidatorPublicKey remains here because we leave the validator struct in tact, later when we remove it we will need another fork
@GalRogozinski please open an issue